### DISCUSSION TITLE
Update PHP on Raspbian for FreshRSS

### DISCUSSION THREAD

**Discussion by LemonsParty** [FreshRSS/FreshRSS#5820] [Q&A] [answered]
Hi all, I see from the release notes for FreshRSS 1.22.1 that it's going to drop support for PHP 7.3 soon. I'm running Raspbian Buster and have PHP 7.3 installed there, so I'd like to get ahead of the game by upgrading so I'm ready to go when 1.23 rolls around. I tried following the instructions [here](https://pimylifeup.com/raspberry-pi-latest-php/), but when I selected "Installation Check" from the FreshRSS web interface, it still showed me running PHP 7.3, even though checking the PHP version from Terminal with "php -v" showed the updated version as expected.

Any idea what went wrong there, or why FreshRSS isn't seeing the updated PHP version? Is there a better way to upgrade PHP?

Note that I had a full backup of my Raspbian SD card and so I restored that to get myself back to where I started just in case my attempt to upgrade had messed something up, which means I'm starting back from a pretty clean slate.

**Comment by Alkarex** [comment, marked as answer]
Hello,
What Web server are you using?
Try `ls /etc/php/` as it might indicate that you have multiple versions installed, and your Web server might not have picked the latest one.

**Reply by Alkarex** [reply]
P.S. We also offer a Docker image, which runs on Raspberry Pi https://hub.docker.com/r/freshrss/freshrss/

**Reply by LemonsParty** [reply]
Thanks for the quick response! I'm using Apache2 as my web server. After updating PHP again, running `ls /etc/php/` did indeed show multiple versions of PHP. I would have expected that, since `php -v` was showing me the newer version, but FreshRSS's interface was still showing the older one. FreshRSS getting its PHP version from Apache (which makes sense!) was the key to unlocking this for me. I followed [these instructions](https://tecadmin.net/switch-between-multiple-php-version-on-debian/) to change the version of PHP being used by Apache, and now FreshRSS is recognizing the newer version in Installation Check as well. Hooray for being future-proofed!

I'm not familiar with Docker, but a quick look seems interesting. I'll definitely check that out as a possibility for the future. Thanks for the heads up, and thanks again for the quick response that pointed me where I needed to go!
